# <span style="color: mediumvioletred;"><u>**Lecture 6 Questions on Introduction to Python Lecture:**</u></span>
#### <span style="color: MediumPurple;">Sophia Dalal </span>
#### <span style="color: MediumPurple;">February 20 2025  </span>

![image.png](https://www.mooc.org/hubfs/python-programming-image-jpg.jpeg)

<span style="color: violet;">**1. Use Python to calculate solutions to the following expressions?**</span>

$\frac{(33+7)}{5}$ and $3 \times 8$

In [1]:
(33+4)/5

7.4

In [2]:
3*8

24

The value of the first expression is 7.4. The value of the second experession is 24. 

<span style="color: violet;"> **2. Determine the data type of the output of the code 1+2. and 1+2. Explain why the data types are different.** </span>

In [5]:
(type(1 + 2.))

float

In [6]:
(type(1 + 2))

int

The data type of the first expression is float. The data type of the second expression is integer. They are different because an integer added to a float creates a float and a integer added to an integer creates a integer. 

<span style="color: violet;"> **3. Generate a loop that prints n^2 for each of the numbers n={1,2,3,4,5}** </span>

In [7]:
for i in {1, 2, 3, 4, 5}:
    print(i**2)

1
4
9
16
25


<span style="color: violet;"> **4. Create the variable i by assigning the value 1 to it. Create a loop that prints the value of i2 to the screen. Increment the value of i by 1 until the value of i2 exceeds 1000. Print the value assigned to i after the loop has completed. Explain why the solution is i=32, where 322 is larger than 1000** </span>

In [9]:
# Create variable i and set it to 1
i = 1

# Loop until i^2 exceeds 1000
while i**2 <= 1000:
    print(i**2)  # Print i squared
    i += 1  # Increment i by 1

# Print the final value of i after the loop terminates
print("Final value of i:", i)

1
4
9
16
25
36
49
64
81
100
121
144
169
196
225
256
289
324
361
400
441
484
529
576
625
676
729
784
841
900
961
Final value of i: 32


The loop stops once i^2 > 1000, meaning the last value assigned to i after the loop is 32.
The loop does not print 1024 because it checks the condition before printing.
The last valid i^2 printed was 31^2 = 961, but when i = 32, the loop stops before printing 32^2 = 1024.
This is why the final assigned value of i is 32, even though 32^2 is greater than 1000.